src/main/java/com/aws/greengrass/deployment/ShadowDeploymentListener.java

```diff
@@ -100,15 +112,68 @@ public void onConnectionResumed(boolean sessionPresent) {
     private final AtomicReference<Map<String, Object>> lastDeploymentStatus = new AtomicReference();
     protected static final Random JITTER = new Random();
 
+    /**
+     * Constructor for unit testing.
+     * @param deploymentQueue {@link DeploymentQueue}
+     * @param deploymentStatusKeeper {@link DeploymentStatusKeeper}
+     * @param mqttClient {@link MqttClient}
+     * @param executorService {@link ExecutorService}
+     * @param deviceConfiguration {@link DeviceConfiguration}
+     * @param iotShadowClient {@link IotShadowClient}
+     */
+    public ShadowDeploymentListener(DeploymentQueue deploymentQueue, DeploymentStatusKeeper deploymentStatusKeeper,
+                                    MqttClient mqttClient, ExecutorService executorService,
+                                    DeviceConfiguration deviceConfiguration, IotShadowClient iotShadowClient) {
+        this.deploymentQueue = deploymentQueue;
+        this.deploymentStatusKeeper = deploymentStatusKeeper;
+        this.mqttClient = mqttClient;
+        this.executorService = executorService;
+        this.deviceConfiguration = deviceConfiguration;
+        this.iotShadowClient = iotShadowClient;
+    }
+
     @Override
     public void postInject() {
-        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
-            logger.atWarn().log("Device not configured to talk to AWS Iot cloud. Single device deployment is offline");
+        deviceConfiguration.onAnyChange((what, node) -> {
+            if (WhatHappened.childChanged.equals(what) && node != null && relevantNodeChanged(node)) {
+                try {
+                    connectToShadowService(deviceConfiguration);
+                } catch (DeviceConfigurationException e) {
+                    logger.atWarn().log("Device not configured to talk to AWS Iot cloud. "
+                            + "Single device deployment is offline: {}", e.getMessage());
+                    return;
+                }
+            }
+        });
+
+        try {
+            connectToShadowService(deviceConfiguration);
```

Comment:
probably same issue as jobs. Make sure we don't have multiple threads trying to connect.